### PR TITLE
cluster-monitoring: Fix incorrect handling of conditional PVCs

### PR DESCRIPTION
**Pull request by brancz**
Closes https://bugzilla.redhat.com/show_bug.cgi?id=1623786

cc @juzhao @sdodson @vrutkovs

### CODE REVIEW

**Comment by brancz** [comment]
This needs to be cherry-picked into the 3.11 release

File: roles/openshift_cluster_monitoring_operator/templates/cluster-monitoring-operator-config.j2 [hunk 1]
```diff
@@ -25,7 +25,7 @@ data:
 {% endif %}
       externalLabels:
         cluster: {{ openshift_cluster_monitoring_operator_cluster_id }}
-{% if openshift_cluster_monitoring_operator_prometheus_storage_enabled %}
+{% if openshift_cluster_monitoring_operator_prometheus_storage_enabled | bool %}
```
Comment (squat) [inline roles/openshift_cluster_monitoring_operator/templates/cluster-monitoring-operator-config.j2]: nice

**Pull request merged**